oecologia-australis.csl

```diff
@@ -268,42 +268,54 @@
           <choose>
             <if match="any" variable="number-of-pages publisher">
               <label variable="number-of-pages" form="short" plural="never"/>
-              <text variable="number-of-pages"/>
+              <text variable="number-of-pages" suffix="."/>
               <text variable="publisher" prefix="ed. " suffix=", "/>
               <text variable="publisher-place"/>
             </if>
           </choose>
         </group>
       </if>
       <else-if type="thesis" match="any">
+        <group suffix=".">
+          <text variable="genre" prefix=" "/>
+        </group>
         <group delimiter=", ">
           <text variable="publisher-place"/>
-          <text variable="publisher"/>
+          <text variable="publisher" prefix=" "/>
```

Comment:
no need for the prefix here. It's already in the delimiter above, but I suspect you want to set in on the group (i.e. line 280)

Reply:
Isn´t it on a group? I didn´t understand the comment.

Reply:
no you have `prefix=" "` on the publisher variable. But you already have a space between `publisher-place` and `publisher` because of the `delimiter=", "` in the group. 

On the other hand, you don't have, best I can tell, a space before `publisher-place` at all. So you should move `prefix=" "` to the group two lines above rather than keeping it on the variable to _both_ avoid the duplicate space (even if Zotero will strip that ) and to make sure the place of publication has a space in front of it when it's present. I'm suspecting you tested this with thesis data without a space.

Reply:
Ok, I will exclude it.